data/scripts/creaturescripts/bestiary_kill.lua

```diff
@@ -0,0 +1,23 @@
+local bestiaryOnKill = CreatureEvent("BestiaryOnKill")
+function bestiaryOnKill.onKill(player, creature, lastHit)
+	if not player:isPlayer() or not creature:isMonster() or creature:hasBeenSummoned() or creature:isPlayer() then
+		return true
+	end
+
+	for cid, damage in pairs(creature:getDamageMap()) do
+		local participant = Player(cid)
+		if participant and participant:isPlayer() then
+			participant:addBestiaryKill(creature:getName())
+		end
+	  end
+
+	return true
+end
+bestiaryOnKill:register()
+
+local loginBestiaryPlayer = CreatureEvent("loginBestiaryPlayer")
+function loginBestiaryPlayer.onLogin(player)
+	player:registerEvent("BestiaryOnKill")
+	return true
+end
+loginBestiaryPlayer:register()
```

Comment:
Fix indentation to tab and corpse of "for" wrong indentation
```suggestion
local bestiaryOnKill = CreatureEvent("BestiaryOnKill")

function bestiaryOnKill.onKill(player, creature, lastHit)
	if not player:isPlayer() or not creature:isMonster() or creature:hasBeenSummoned() or creature:isPlayer() then
		return true
	end

	for cid, damage in pairs(creature:getDamageMap()) do
		local participant = Player(cid)
		if participant and participant:isPlayer() then
			participant:addBestiaryKill(creature:getName())
		end
	end

	return true
end

bestiaryOnKill:register()

local loginBestiaryPlayer = CreatureEvent("loginBestiaryPlayer")

function loginBestiaryPlayer.onLogin(player)
	player:registerEvent("BestiaryOnKill")
	return true
end

loginBestiaryPlayer:register()
```